File: 2015/day01/typescript/2015_day01.ts
import { readFileSync } from 'fs';

const steps: string = readFileSync('../input.txt', 'utf8')

let floor: number = 0;

for (let step of steps) {
    if (step == "(") {
        floor += 1;
    } else if (step == ")") {
        floor -= 1;
    } else {
        continue
    }
}

console.log(`Santa ends up at floor: ${floor}`);

File: 2015/day01/go/2015_day01.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"log"
	"os"
)

func main() {
	file, err := os.Open("../input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	reader := bufio.NewReader(file)
	floor := 0

	for {
		char, _, err := reader.ReadRune()
		if err == io.EOF {
			break
		}
		if err != nil {
			log.Fatal(err)
		}

		switch char {
		case '(':
			floor++
		case ')':
			floor--
		}
	}

	fmt.Println("Santa ends at floor:", floor)
}
